### PR TITLE
#94 create user request

### LINKED ISSUE #94: VB-11: Add endpoint for creating a new Request
POST /api/requests
Media type: Application/json

RequestBody(required): Request object as json

Responses:
201: Request object created
Example value:
```
{
  "beneficiary": "beneficiary_id",
  "user": "user_id",
  "status": "new",
  "secret": "string",
  "urgent": false,
  "comments": "string",
  "has_symptoms": false,
  "cluster": "cluster_id",
  "created_at": "2020-09-10T21:31:16.741Z"
}
```
400: Bad request
Example value:
```
{
  "code": "string",
  "message": "string"
}
```
401: Unauthorized Access
Example value:
`Unauthorized Access`

### PR DESCRIPTION
**Pull request by sroy8091**
### Requirements for making a pull request

Thank you for contributing to our project! 

Please fill out the template below to help the project maintainers review it as fast as possible and include your contribution to the project.

### What does it fix?

Closes #94 

Please mention the main changes this PR brings.

### How has it been tested?

Please describe the tests that you ran to verify your changes.

### CODE REVIEW

**Comment by johnsmithm** [comment]
can you also add info to the /backend/static/swagger.yaml and user request examples in the backend/seeds.py?

File: backend/endpoints/user_request.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    -------
+    200:
+        If the request was successfully created and saved.
+    400:
+        If the request wasn't created or saved, and there was raised some exception.
+    """
+    try:
+        user_request_data = request_json
+        log.debug(request_json)
+        user_request = Request(**user_request_data)
```
Comment (johnsmithm) [inline backend/endpoints/user_request.py:37]:
```
                  "user": 123456,
                  "created_at": "2020-10-04T21:31:16.741Z"
```
@alexciornii @iamandrewluca is it better to add these fields in the backend? not to be send from frontend?

Comment (alexciornii) [reply backend/endpoints/user_request.py:37]: it's not necessary, because this fields are filling automatic

File: backend/app.py [hunk 2]
```diff
@@ -158,6 +159,13 @@ def get_beneficiary_by_filters(pages=15, per_page=10):
     return get_beneficiaries_by_filters(request.args, pages, per_page)
 
 
+# user request
+@app.route("/api/request", methods=["POST"])
+# @auth.login_required
```
Comment (alexciornii) [inline backend/app.py:164]: login is required

**Review by alexciornii — CHANGES_REQUESTED** [review]

File: backend/endpoints/user_request.py (hunk 1 above)
Comment (johnsmithm) [reply backend/endpoints/user_request.py:37]: @sroy8091  use the` auth.username()` as [here](https://github.com/code4moldova/voluntar-backend/blob/development/backend/app.py#L100)

**Comment by johnsmithm** [comment]
will you be able to add swagger and created_by from auth this weekend? @sroy8091

**Comment by sroy8091** [comment]
yes, sure @johnsmithm

File: backend/endpoints/user_request.py (hunk 1 above)
Comment (sroy8091) [reply backend/endpoints/user_request.py:37]: @johnsmithm so basically instead of taking user from body we will populate according to user requesting?

**Comment by sroy8091** [comment]
@johnsmithm it seems request schema and api details are already added in swagger.yaml

**Comment by johnsmithm** [comment]
@sroy8091 request schema and api is added but need to be corrected. for the post method there is no `request_id`. the value for `user` property must be the a `user_id` example

File: backend/endpoints/user_request.py (hunk 1 above)
Comment (sroy8091) [reply backend/endpoints/user_request.py:37]: @johnsmithm can you also confirm this?

Comment (johnsmithm) [reply backend/endpoints/user_request.py:37]: yes

**Comment by johnsmithm** [comment]
@sroy8091 any updates?

File: backend/app.py [hunk 3]
```diff
@@ -158,6 +159,14 @@ def get_beneficiary_by_filters(pages=15, per_page=10):
     return get_beneficiaries_by_filters(request.args, pages, per_page)
 
 
+# user request
+@app.route("/api/request", methods=["GET"])
```
Comment (vitaliel) [inline backend/app.py:163]:
```suggestion
@app.route("/api/request", methods=["POST"])
```

File: backend/app.py [hunk 4]
```diff
@@ -158,6 +159,14 @@ def get_beneficiary_by_filters(pages=15, per_page=10):
...
+@auth.login_required
+def new_user_request():
+    print("here")
```
Comment (vitaliel) [inline backend/app.py:166]: Remove print line

Comment (sroy8091) [reply backend/app.py]: done

File: backend/app.py (hunk 3 above)
Comment (sroy8091) [reply backend/app.py]: done

**Comment by sroy8091** [comment]
Thanks @johnsmithm for accepting my pr. I would love to continue contributing to this repo.

**Comment by johnsmithm** [comment]
thank you for your contribution, comment on some issue and I will assign it to you

**Pull request merged**